htmresearch/algorithms/faulty_temporal_memory.py

```diff
@@ -70,206 +167,139 @@ def killCells(self, percent = 0.05):
     print "Total number of dead cells=",len(self.deadCells)
 
     numSegmentDeleted = 0
-    for cell in self.deadCells:
-      segmentsPerCell = list(self.connections.segmentsForCell(cell))
-      for segment in segmentsPerCell:
-        self.connections.destroySegment(segment)
-        numSegmentDeleted += 1
-
-    print "Total number of segments removed=", numSegmentDeleted
-
-
-  def activateCorrectlyPredictiveCells(self,
-                                       prevPredictiveCells,
-                                       prevMatchingCells,
-                                       activeColumns):
-    """
-    Phase 1: Activate the correctly predictive cells.
-
-    Pseudocode:
-
-      - for each prev predictive cell
-        - if in active column
-          - mark it as active
-          - mark it as winner cell
-          - mark column as predicted => active
-        - if not in active column
-          - mark it as an predicted but inactive cell
-
-    @param prevPredictiveCells (set) Indices of predictive cells in `t-1`
-    @param activeColumns       (set) Indices of active columns in `t`
-
-    @return (tuple) Contains:
-                      `activeCells`               (set),
-                      `winnerCells`               (set),
-                      `predictedActiveColumns`    (set),
-                      `predictedInactiveCells`    (set)
-    """
-    activeCells = set()
-    winnerCells = set()
-    predictedActiveColumns = set()
-    predictedInactiveCells = set()
-
-    prevPredictiveCells = prevPredictiveCells - self.deadCells
-    prevMatchingCells = prevMatchingCells - self.deadCells
-
-    for cell in prevPredictiveCells:
-      column = self.columnForCell(cell)
 
-      if column in activeColumns:
-        activeCells.add(cell)
-        winnerCells.add(cell)
-        predictedActiveColumns.add(column)
+    for cellIdx in self.deadCells:
+      numSegmentDeleted += self.connections.destroyCell(cellIdx)
 
-    if self.predictedSegmentDecrement > 0:
-      for cell in prevMatchingCells:
-        column = self.columnForCell(cell)
+    print "Total number of segments removed=", numSegmentDeleted
 
-        if column not in activeColumns:
-          predictedInactiveCells.add(cell)
 
-    return (activeCells,
-            winnerCells,
-            predictedActiveColumns,
-            predictedInactiveCells)
+  def activatePredictedColumn(self, *args, **kwargs):
```

Comment:
`activatePredictedColumn` won't activate any dead cells, since dead cells can't have active segments. So I don't think this method is needed.